### PR TITLE
Allow merge from --postman and --har with --config

### LINKED ISSUE #422: Take probes from --postman and --har to merge with --config
For now, flag `--postman` and `--har` are exclusive against `-c/--config`.
Considering `--postman/--har` only parse file as probe collection, it will be convenient to merge it with all properties but `probes` from `-c/--config`.

### PR DESCRIPTION
**Pull request by samsunoor**
# Monika Pull Request (PR) 
This PR resolves #422 

## What feature/issue does this PR add  
1. Allow `--har` and `--postman` to be combined with `-c/--config`

## How did you implement / how did you fix it  
1.  Probes from `har` and `postman` file will always override configurations passed to `--config`
2. When no additional config provided, `desktop` notification type will be default
3. `--har` and `--postman` are still exclusive against each other

## How to test  
- `npm run -- -c /path/to/notifications/config --har test/testConfigs/harTest.har`
- `npm run -- -c /path/to/notifications/config --postman test/testConfigs/simple.postman_collection.json`

### CODE REVIEW

File: docs/src/pages/guides/cli-options.md [hunk 1]
```diff
@@ -76,17 +76,32 @@ The example above flushes the database bypassing without waiting for user confir
...
+
+### Merge HAR file to existing configurations
+
+You can also use `-c/--config` to merge properties with them. Note that using `--har` will override probes passed to `-c/--config`.
+
+```bash
+monika --config monika-notifications.json -H my-file.har
+```
+
+**P.S.**: HAR files may contain sensitive information, use caution when distributing HAR filles.
```
Comment (gsukanto) [inline docs/src/pages/guides/cli-options.md:104]: `use with caution`

File: docs/src/pages/guides/cli-options.md [hunk 2]
```diff
@@ -76,17 +76,32 @@ The example above flushes the database bypassing without waiting for user confir
 
 ## HAR
 
-Monika supports HAR files as input. HAR are JSON formatted HTTP ARchive file. Generate a HAR file from the site you've visited then use Monika to refetch the pages and ensure they still work.
+Monika supports HAR files as input. HAR are JSON formatted HTTP ARchive file. Follow the steps [here](https://medium.com/hyperjump-tech/generate-your-monika-configuration-using-http-archive-har-764944cbb9e6) to generate your own HAR file from the site you've visited then use Monika to refetch the pages and ensure they still work.
```
Comment (gsukanto) [inline docs/src/pages/guides/cli-options.md:79]: `Follow these steps`

File: src/components/config/index.ts [hunk 3]
```diff
@@ -135,16 +135,35 @@ const setupConfigFromJson = (flags: any): Promise<Partial<Config>>[] => {
...
-  if (flags.har) {
-    configParse.push(parseConfig(flags.har, 'har'))
-  } else if (flags.postman) {
-    configParse.push(parseConfig(flags.postman, 'postman'))
-  } else if (Array.isArray(flags.config) && flags.config.length > 0) {
+  if (Array.isArray(flags.config) && flags.config.length > 0) {
     const json = setupConfigFromJson(flags)
     configParse.push(...json)
   }
+  if (flags.har) {
```
Comment (gsukanto) [inline src/components/config/index.ts:154]: I think we can make this code more readable. For example:
```
  let parsed
  if (flags.har) {
    parsed = parseConfig(flags.har, 'har')
  } else if (flags.postman) {
    parsed = parseConfig(flags.postman, 'postman')
  }

  if (configParse.length === 0) {
    parsed = addDefaultNotifications(parsed)
  }
  configParse.push(parsed)
```

**Review by gsukanto — APPROVED** [review]

File: docs/src/pages/guides/cli-options.md (hunk 1 above)
Comment (samsunoor) [reply docs/src/pages/guides/cli-options.md:104]: I guess this is still acceptable, though.
*[reference](https://www.merriam-webster.com/dictionary/caution)*

**Review by nicnocquee — APPROVED** [review]

**Pull request merged**